File: playwright/pages/cart.page.ts
import { Locator, Page } from '@playwright/test'

export class CartPage {
    readonly page: Page;
    readonly lista: Locator;
    readonly itens: Locator;
    readonly botaoCheckout: Locator;

    constructor(page: Page) {
        this.page = page;
        this.lista = page.locator('[data-test="cart-list"]');
        this.itens = this.lista.locator('[data-test="inventory-item"]');
        this.botaoCheckout = page.locator('[data-test="checkout"]');
    }

    async irParaCheckout(): Promise<void> {
        await this.botaoCheckout.click();
    }
}

File: selenium/src/test/java/pages/CartPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class CartPage extends BasePage {

    private static final By LISTA_DO_CARRINHO = By.cssSelector("[data-test='cart-list']");
    private static final By ITEM_DO_CARRINHO = By.cssSelector("[data-test='inventory-item']");
    private static final By BOTAO_CHECKOUT = By.cssSelector("[data-test='checkout']");

    public CartPage(WebDriver driver) {
        super(driver);
    }

    public int quantidadeDeItens() {
        visivel(LISTA_DO_CARRINHO);
        return driver.findElements(ITEM_DO_CARRINHO).size();
    }

    public CheckoutPage irParaCheckout() {
        clicar(BOTAO_CHECKOUT);
        return new CheckoutPage(driver);
    }
}
